// annar/esm/cell/index.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _typeof from "@babel/runtime/helpers/esm/typeof";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import * as React from 'react';
import { View, Text } from 'remax/one';
import classNames from 'classnames';
import Icon from '../icon';
import FormValue from '../form-value';
import Picker from '../picker';
import Input from '../input';
import { getPrefixCls } from '../common';
var prefixCls = getPrefixCls('cell');

var Cell = function Cell(props) {
  var label = props.label,
      style = props.style,
      labelStyle = props.labelStyle,
      valueStyle = props.valueStyle,
      children = props.children,
      description = props.description,
      verticality = props.verticality,
      icon = props.icon,
      _props$border = props.border,
      border = _props$border === void 0 ? true : _props$border,
      arrow = props.arrow,
      extra = props.extra,
      required = props.required,
      field = props.field,
      _props$defaultNullVal = props.defaultNullValue,
      defaultNullValue = _props$defaultNullVal === void 0 ? '' : _props$defaultNullVal,
      onTap = props.onTap;

  if (verticality) {
    return /*#__PURE__*/React.createElement(View, {
      className: prefixCls,
      style: style,
      onTap: onTap
    }, /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-verticality")
    }, label ? /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-verticality-label"),
      style: labelStyle
    }, required ? /*#__PURE__*/React.createElement(Text, {
      className: "".concat(prefixCls, "-verticality-required")
    }, "*") : null, label) : null, /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-verticality-value"),
      style: valueStyle
    }, children)));
  }

  return /*#__PURE__*/React.createElement(View, {
    className: prefixCls,
    style: style,
    onTap: onTap
  }, /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-container")
  }, /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-container-main")
  }, /*#__PURE__*/React.createElement(View, {
    className: classNames("".concat(prefixCls, "-container-main-left"), _defineProperty({}, "".concat(prefixCls, "-container-main-left-input"), field))
  }, /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-container-main-left-label")
  }, required ? /*#__PURE__*/React.createElement(Text, {
    className: "".concat(prefixCls, "-container-main-left-label-required")
  }, "*") : null, icon ? /*#__PURE__*/React.createElement(Icon, {
    type: icon,
    size: "32px",
    style: {
      marginRight: '10px'
    },
    color: "#333"
  }) : null, label ? /*#__PURE__*/React.createElement(Text, {
    className: "".concat(prefixCls, "-container-main-left-label-value"),
    style: labelStyle
  }, label) : null)), /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-container-main-right")
  }, /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-container-main-right-value"),
    style: valueStyle
  }, children || children === 0 ? children : defaultNullValue), extra ? /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-extra")
  }, extra) : null, arrow ? /*#__PURE__*/React.createElement(Icon, {
    type: "right",
    style: {
      marginLeft: '10px'
    },
    color: "#666"
  }) : null)), description ? /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-container-description")
  }, description) : null), border ? /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-border")
  }) : null);
};

Cell.Picker = function (props) {
  var _props$align = props.align,
      align = _props$align === void 0 ? 'left' : _props$align,
      value = props.value,
      onChange = props.onChange,
      placeholder = props.placeholder,
      children = props.children,
      disabled = props.disabled,
      range = props.range,
      _props$rangeKey = props.rangeKey,
      rangeKey = _props$rangeKey === void 0 ? 'text' : _props$rangeKey;
  var selectedText = '';

  if (Array.isArray(value)) {
    var selected = value.map(function (i, columnIndex) {
      var _range$columnIndex;

      return range === null || range === void 0 ? void 0 : (_range$columnIndex = range[columnIndex]) === null || _range$columnIndex === void 0 ? void 0 : _range$columnIndex[i];
    });
    selectedText = selected.map(function (i) {
      return _typeof(i) === 'object' ? i[rangeKey] || '' : i;
    }).join(' ');
  } else {
    var _selected = range === null || range === void 0 ? void 0 : range[value];

    selectedText = _typeof(_selected) === 'object' ? _selected[rangeKey] || '' : _selected;
  }

  return /*#__PURE__*/React.createElement(Cell, _extends({
    labelStyle: {
      width: '180px'
    }
  }, props), /*#__PURE__*/React.createElement(Picker, {
    range: range,
    rangeKey: rangeKey,
    disabled: disabled,
    value: value,
    onChange: onChange
  }, children !== null && children !== void 0 ? children : /*#__PURE__*/React.createElement(FormValue, {
    placeholder: placeholder,
    style: {
      textAlign: align,
      color: disabled ? '#999' : undefined
    }
  }, selectedText)));
};

var defaultInputWrapperStyle = {
  display: 'flex',
  padding: 0,
  border: 0
};
var defaultInputStyle = {
  height: '48px',
  lineHeight: '48px',
  minHeight: '48px'
};

Cell.Input = function (props) {
  return /*#__PURE__*/React.createElement(Cell, _extends({
    field: true,
    labelStyle: {
      width: '180px'
    }
  }, props), /*#__PURE__*/React.createElement(Input, _extends({}, props, {
    style: defaultInputWrapperStyle,
    inputStyle: defaultInputStyle
  })));
};

export default Cell;
