// annar/web/picker/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _web = require("../web");

var _popup = _interopRequireDefault(require("../popup"));

var _pickerView = _interopRequireDefault(require("../picker-view"));

var _pickerViewColumn = _interopRequireDefault(require("../picker-view-column"));

var _common = require("../common");

var prefixCls = (0, _common.getPrefixCls)('picker');

var Picker = function Picker(props) {
  var _props$value = props.value,
      value = _props$value === void 0 ? 0 : _props$value,
      range = props.range,
      _props$rangeKey = props.rangeKey,
      rangeKey = _props$rangeKey === void 0 ? 'text' : _props$rangeKey,
      disabled = props.disabled,
      children = props.children,
      onChange = props.onChange;

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      open = _useState2[0],
      setOpen = _useState2[1];

  var val = (0, _react.useRef)(value < 0 ? 0 : value);

  var handleTap = function handleTap() {
    if (disabled) {
      return;
    }

    setOpen(function (state) {
      return !state;
    });
  };

  var handleOK = function handleOK(e) {
    onChange === null || onChange === void 0 ? void 0 : onChange(val.current, e);
    setOpen(false);
  };

  var handleChangeColumn = function handleChangeColumn(multiple, v, index) {
    if (multiple) {
      val.current = (0, _toConsumableArray2["default"])(Array.isArray(val.current) ? val.current : [val.current]);
      val.current[index] = v;
      return;
    }

    val.current = v;
  };

  var multiple = false;

  if (Array.isArray(range === null || range === void 0 ? void 0 : range[0])) {
    multiple = true;
  }

  return /*#__PURE__*/_react["default"].createElement(_web.View, {
    className: prefixCls
  }, /*#__PURE__*/_react["default"].createElement(_web.View, {
    onTap: handleTap
  }, children), /*#__PURE__*/_react["default"].createElement(_popup["default"], {
    position: "bottom",
    open: open,
    onClose: function onClose() {
      return setOpen(false);
    }
  }, /*#__PURE__*/_react["default"].createElement(_pickerView["default"], {
    onOK: handleOK,
    onCancel: function onCancel() {
      return setOpen(false);
    }
  }, multiple ? range === null || range === void 0 ? void 0 : range.map(function (columnData, index) {
    return /*#__PURE__*/_react["default"].createElement(_pickerViewColumn["default"], {
      key: index,
      value: value[index],
      options: columnData,
      optionsKey: rangeKey,
      onChange: function onChange(v) {
        return handleChangeColumn(true, v, index);
      }
    });
  }) : /*#__PURE__*/_react["default"].createElement(_pickerViewColumn["default"], {
    value: value,
    options: range,
    optionsKey: rangeKey,
    onChange: function onChange(v) {
      return handleChangeColumn(false, v);
    }
  }))));
};

var _default = Picker;
exports["default"] = _default;
