// annar/esm/grid/index.js
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import React, { useMemo } from 'react';
import { View } from 'remax/one';
import classNames from 'classnames';
import Row from '../row';
import Col from '../col';
import { twoDimensional, flat } from '../_util';
import { getPrefixCls } from '../common';
var prefixCls = getPrefixCls('grid');

var getColIndex = function getColIndex(two, rindex, cindex) {
  var list = two.slice(0, rindex + 1);
  list[rindex] = list[rindex].slice(0, cindex + 1);
  var flatedList = flat(list);
  return flatedList.length;
};

var Grid = function Grid(props) {
  var _props$data = props.data,
      data = _props$data === void 0 ? [] : _props$data,
      _props$columns = props.columns,
      columns = _props$columns === void 0 ? 4 : _props$columns,
      _props$gutter = props.gutter,
      gutter = _props$gutter === void 0 ? 0 : _props$gutter,
      span = props.span,
      border = props.border,
      children = props.children;

  var _useMemo = useMemo(function () {
    return twoDimensional(data, columns, span, 24);
  }, [data === null || data === void 0 ? void 0 : data.length]),
      _useMemo2 = _slicedToArray(_useMemo, 2),
      two = _useMemo2[0],
      spanTwo = _useMemo2[1];

  var gutterArray = gutter;

  if (!Array.isArray(gutter)) {
    gutterArray = [gutter];
  }

  return /*#__PURE__*/React.createElement(View, {
    className: classNames(prefixCls)
  }, two === null || two === void 0 ? void 0 : two.map(function (row, rindex) {
    return /*#__PURE__*/React.createElement(Row, {
      key: rindex,
      gutter: gutterArray[0],
      className: border && "".concat(prefixCls, "-row-border"),
      style: {
        marginBottom: rindex !== two.length - 1 ? "".concat(gutterArray[1], "px") : undefined,
        borderBottom: two.length - 1 !== rindex ? border : undefined
      }
    }, row === null || row === void 0 ? void 0 : row.map(function (col, cindex) {
      var _spanTwo$rindex;

      return /*#__PURE__*/React.createElement(Col, {
        key: cindex,
        span: spanTwo === null || spanTwo === void 0 ? void 0 : (_spanTwo$rindex = spanTwo[rindex]) === null || _spanTwo$rindex === void 0 ? void 0 : _spanTwo$rindex[cindex],
        className: border && "".concat(prefixCls, "-col-border"),
        style: {
          borderRight: row.length - 1 !== cindex ? border : undefined
        }
      }, typeof children === 'function' ? children === null || children === void 0 ? void 0 : children(col, getColIndex(spanTwo, rindex, cindex), cindex, rindex) : children);
    }));
  }));
};

export default Grid;
