// annar/esm/space/index.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import * as React from 'react';
import { View } from 'remax/one';
import classNames from 'classnames';
import { getPrefixCls } from '../common';
var prefixCls = getPrefixCls('space');

var renderItems = function renderItems(children, size, direction) {
  var itemStyle = undefined;

  if (typeof size === 'number') {
    itemStyle = direction === 'vertical' ? {
      marginBottom: size
    } : {
      marginRight: size
    };
  }

  var items = [];
  React.Children.forEach(children, function (child, index) {
    var itemCn = classNames("".concat(prefixCls, "-item"), _defineProperty({}, "".concat(prefixCls, "-").concat(direction, "-").concat(size), typeof size === 'string' && size));

    if (child === undefined || child === null) {
      return;
    }

    items.push( /*#__PURE__*/React.createElement(View, {
      key: index,
      className: itemCn,
      style: itemStyle
    }, child));
  });
  return items;
};

var Space = function Space(props) {
  var _props$size = props.size,
      size = _props$size === void 0 ? 'small' : _props$size,
      _props$direction = props.direction,
      direction = _props$direction === void 0 ? 'horizontal' : _props$direction,
      _props$align = props.align,
      align = _props$align === void 0 ? direction === 'horizontal' ? 'center' : undefined : _props$align,
      style = props.style,
      className = props.className,
      children = props.children;
  var cn = classNames(prefixCls, "".concat(prefixCls, "-").concat(direction), _defineProperty({}, "".concat(prefixCls, "-align-").concat(align), align), className);
  return /*#__PURE__*/React.createElement(View, {
    className: cn,
    style: style
  }, renderItems(children, size, direction));
};

export default Space;
