// annar/web/filter/item.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _web = require("../web");

var _classnames = _interopRequireDefault(require("classnames"));

var _dropdown = _interopRequireDefault(require("../dropdown"));

var _popup = _interopRequireDefault(require("../popup"));

var _common = require("../common");

var prefixCls = (0, _common.getPrefixCls)('filter-item');

var Item = function Item(props, ref) {
  var _classNames, _options$find;

  var title = props.title,
      onTap = props.onTap,
      children = props.children,
      onShowMask = props.onShowMask,
      value = props.value,
      _props$options = props.options,
      options = _props$options === void 0 ? [] : _props$options,
      onChange = props.onChange,
      activeColor = props.activeColor,
      disabled = props.disabled;

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      open = _useState2[0],
      setOpen = _useState2[1];

  (0, _react.useImperativeHandle)(ref, function () {
    return {
      toggle: function toggle() {
        setOpen(function (state) {
          if (state) {
            setTimeout(function () {
              onShowMask === null || onShowMask === void 0 ? void 0 : onShowMask(false);
            });
          } else {
            setTimeout(function () {
              onShowMask === null || onShowMask === void 0 ? void 0 : onShowMask(true);
            });
          }

          return !state;
        });
      },
      open: function open() {
        setOpen(true);
      },
      close: function close() {
        setOpen(false);
      }
    };
  }, []);

  var handleTap = function handleTap() {
    if (disabled) {
      return;
    }

    setOpen(function (state) {
      return !state;
    });
    onTap === null || onTap === void 0 ? void 0 : onTap(!open);
  };

  var handleChange = function handleChange(e) {
    onChange === null || onChange === void 0 ? void 0 : onChange(e);
    setOpen(false);
    onShowMask === null || onShowMask === void 0 ? void 0 : onShowMask(false);
  };

  return /*#__PURE__*/_react["default"].createElement(_web.View, {
    className: prefixCls
  }, /*#__PURE__*/_react["default"].createElement(_web.View, {
    style: {
      backgroundColor: '#FDFFFD'
    }
  }, /*#__PURE__*/_react["default"].createElement(_web.View, {
    className: (0, _classnames["default"])((_classNames = {}, (0, _defineProperty2["default"])(_classNames, "".concat(prefixCls, "-label"), true), (0, _defineProperty2["default"])(_classNames, "".concat(prefixCls, "-label-disabled"), disabled), _classNames)),
    onTap: handleTap
  }, /*#__PURE__*/_react["default"].createElement(_web.Text, {
    className: "".concat(prefixCls, "-label-text")
  }, (options === null || options === void 0 ? void 0 : (_options$find = options.find(function (option) {
    return option.value === value;
  })) === null || _options$find === void 0 ? void 0 : _options$find.text) || title), open ? /*#__PURE__*/_react["default"].createElement(_web.Text, {
    className: "".concat(prefixCls, "-label-chevronUp")
  }) : /*#__PURE__*/_react["default"].createElement(_web.Text, {
    className: "".concat(prefixCls, "-label-chevronDown")
  }))), /*#__PURE__*/_react["default"].createElement(_popup["default"], {
    position: "bottom",
    square: true,
    curve: "ease",
    mask: false,
    open: open,
    style: {
      position: 'absolute',
      zIndex: -1,
      transform: open ? 'translate3d(0, 100%, 0)' : 'translate3d(0, 0, 0)'
    }
  }, /*#__PURE__*/_react["default"].createElement(_web.ScrollView, {
    scrollY: true,
    className: (0, _classnames["default"])((0, _defineProperty2["default"])({}, "".concat(prefixCls, "-value"), true))
  }, children ? children : /*#__PURE__*/_react["default"].createElement(_dropdown["default"], {
    value: value,
    options: options,
    onChange: handleChange,
    activeColor: activeColor
  }))));
};

var _default = /*#__PURE__*/_react["default"].forwardRef(Item);

exports["default"] = _default;
