// @remax/cli/lib/build/utils/output.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.output = void 0;
const node_notifier_1 = __importDefault(require("node-notifier"));
const COLORS = {
    red: '\x1b[31m',
    green: '\x1b[32m',
    yellow: '\x1b[33m',
    blue: '\x1b[34m',
};
const RESET = '\x1b[0m';
const output = (content, color, notify = false) => {
    const message = Array.isArray(content) ? content : [content];
    console.log(`${COLORS[color]}%s${RESET}`, ...message);
    if (notify) {
        node_notifier_1.default.notify({
            title: 'remax',
            message: message.join(' '),
        });
    }
};
exports.output = output;
exports.default = {
    message: exports.output,
    error: (message, notify) => exports.output(`\n🚨 ${message}`, 'red', notify),
    warn: (message, notify) => exports.output(`\n⚠️ ${message}`, 'yellow', notify),
};
