// annar/esm/card/index.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import * as React from 'react';
import { View } from 'remax/one';
import classNames from 'classnames';
import { getPrefixCls } from '../common';
var prefixCls = getPrefixCls('card');

var Card = function Card(props) {
  var _classNames3;

  var title = props.title,
      description = props.description,
      extra = props.extra,
      cover = props.cover,
      foot = props.foot,
      _props$direction = props.direction,
      direction = _props$direction === void 0 ? 'vertical' : _props$direction,
      shadow = props.shadow,
      style = props.style,
      contentStyle = props.contentStyle,
      children = props.children,
      onTap = props.onTap;

  var renderTitle = function renderTitle(isHead) {
    return title || description || extra ? /*#__PURE__*/React.createElement(View, {
      className: classNames("".concat(prefixCls, "-head"), _defineProperty({}, "".concat(prefixCls, "-content-head"), !isHead))
    }, /*#__PURE__*/React.createElement(View, null, title ? /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-title")
    }, title) : null, description ? /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-description")
    }, description) : null), /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-extra")
    }, extra)) : null;
  };

  var renderHead = function renderHead() {
    if (cover) {
      return /*#__PURE__*/React.createElement(View, {
        className: "".concat(prefixCls, "-cover")
      }, cover);
    }

    return renderTitle(true);
  };

  var renderFoot = function renderFoot() {
    return foot ? /*#__PURE__*/React.createElement(View, {
      className: classNames("".concat(prefixCls, "-foot"), _defineProperty({}, "".concat(prefixCls, "-foot-horizontal"), direction === 'horizontal'))
    }, foot) : null;
  };

  return /*#__PURE__*/React.createElement(View, {
    className: classNames(prefixCls, (_classNames3 = {}, _defineProperty(_classNames3, "".concat(prefixCls, "-horizontal"), cover && direction === 'horizontal'), _defineProperty(_classNames3, "".concat(prefixCls, "-shadow"), shadow), _classNames3)),
    style: style,
    onTap: onTap
  }, renderHead(), /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-content"),
    style: _objectSpread({}, contentStyle)
  }, cover ? renderTitle(false) : null, children, direction === 'horizontal' ? renderFoot() : null), direction !== 'horizontal' ? renderFoot() : null);
};

export default Card;
