// annar/web/grid/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _web = require("../web");

var _classnames = _interopRequireDefault(require("classnames"));

var _row = _interopRequireDefault(require("../row"));

var _col = _interopRequireDefault(require("../col"));

var _util = require("../_util");

var _common = require("../common");

var prefixCls = (0, _common.getPrefixCls)('grid');

var getColIndex = function getColIndex(two, rindex, cindex) {
  var list = two.slice(0, rindex + 1);
  list[rindex] = list[rindex].slice(0, cindex + 1);
  var flatedList = (0, _util.flat)(list);
  return flatedList.length;
};

var Grid = function Grid(props) {
  var _props$data = props.data,
      data = _props$data === void 0 ? [] : _props$data,
      _props$columns = props.columns,
      columns = _props$columns === void 0 ? 4 : _props$columns,
      _props$gutter = props.gutter,
      gutter = _props$gutter === void 0 ? 0 : _props$gutter,
      span = props.span,
      border = props.border,
      children = props.children;

  var _useMemo = (0, _react.useMemo)(function () {
    return (0, _util.twoDimensional)(data, columns, span, 24);
  }, [data === null || data === void 0 ? void 0 : data.length]),
      _useMemo2 = (0, _slicedToArray2["default"])(_useMemo, 2),
      two = _useMemo2[0],
      spanTwo = _useMemo2[1];

  var gutterArray = gutter;

  if (!Array.isArray(gutter)) {
    gutterArray = [gutter];
  }

  return /*#__PURE__*/_react["default"].createElement(_web.View, {
    className: (0, _classnames["default"])(prefixCls)
  }, two === null || two === void 0 ? void 0 : two.map(function (row, rindex) {
    return /*#__PURE__*/_react["default"].createElement(_row["default"], {
      key: rindex,
      gutter: gutterArray[0],
      className: border && "".concat(prefixCls, "-row-border"),
      style: {
        marginBottom: rindex !== two.length - 1 ? "".concat(gutterArray[1], "px") : undefined,
        borderBottom: two.length - 1 !== rindex ? border : undefined
      }
    }, row === null || row === void 0 ? void 0 : row.map(function (col, cindex) {
      var _spanTwo$rindex;

      return /*#__PURE__*/_react["default"].createElement(_col["default"], {
        key: cindex,
        span: spanTwo === null || spanTwo === void 0 ? void 0 : (_spanTwo$rindex = spanTwo[rindex]) === null || _spanTwo$rindex === void 0 ? void 0 : _spanTwo$rindex[cindex],
        className: border && "".concat(prefixCls, "-col-border"),
        style: {
          borderRight: row.length - 1 !== cindex ? border : undefined
        }
      }, typeof children === 'function' ? children === null || children === void 0 ? void 0 : children(col, getColIndex(spanTwo, rindex, cindex), cindex, rindex) : children);
    }));
  }));
};

var _default = Grid;
exports["default"] = _default;
