// annar/web/popup/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var React = _interopRequireWildcard(require("react"));

var _web = require("../web");

var _classnames = _interopRequireDefault(require("classnames"));

var _mask = _interopRequireDefault(require("../mask"));

var _icon = _interopRequireDefault(require("../icon"));

var _common = require("../common");

var prefixCls = (0, _common.getPrefixCls)('popup');

var Popup = function Popup(props) {
  var _classNames;

  var _props$open = props.open,
      open = _props$open === void 0 ? false : _props$open,
      children = props.children,
      onClose = props.onClose,
      closeable = props.closeable,
      title = props.title,
      _props$position = props.position,
      position = _props$position === void 0 ? 'center' : _props$position,
      style = props.style,
      _props$mask = props.mask,
      mask = _props$mask === void 0 ? true : _props$mask,
      square = props.square,
      curve = props.curve,
      transparent = props.transparent;

  var handleClickMask = function handleClickMask() {
    onClose === null || onClose === void 0 ? void 0 : onClose();
  };

  var classes = (0, _classnames["default"])((_classNames = {}, (0, _defineProperty2["default"])(_classNames, "".concat(prefixCls, "-container"), true), (0, _defineProperty2["default"])(_classNames, "".concat(prefixCls, "-container-").concat(position), position), (0, _defineProperty2["default"])(_classNames, "".concat(prefixCls, "-container-ease"), curve === 'ease'), (0, _defineProperty2["default"])(_classNames, "".concat(prefixCls, "-container-square"), square), (0, _defineProperty2["default"])(_classNames, "".concat(prefixCls, "-container-active"), open), (0, _defineProperty2["default"])(_classNames, "".concat(prefixCls, "-container-transparent"), transparent), _classNames));
  return /*#__PURE__*/React.createElement(_web.View, {
    className: prefixCls
  }, /*#__PURE__*/React.createElement(_web.View, {
    className: classes,
    style: style
  }, title ? /*#__PURE__*/React.createElement(_web.View, {
    className: "".concat(prefixCls, "-container-title")
  }, title) : null, children, closeable ? /*#__PURE__*/React.createElement(_web.View, {
    className: "".concat(prefixCls, "-container-close"),
    onTap: function onTap() {
      onClose === null || onClose === void 0 ? void 0 : onClose();
    }
  }, /*#__PURE__*/React.createElement(_icon["default"], {
    type: "close",
    size: "40px",
    color: "#999"
  })) : null), mask ? /*#__PURE__*/React.createElement(_mask["default"], {
    show: open,
    onTap: handleClickMask
  }) : null);
};

var _default = Popup;
exports["default"] = _default;
