// annar/esm/web/base/picker/index.js
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import React, { useState, useEffect } from 'react';
import Popup from '../../../popup';
import { getPrefixCls } from '../../../common';
var prefixCls = getPrefixCls('base-picker');
var heightUnit = 44;

var Picker = function Picker(props) {
  var _props$value = props.value,
      value = _props$value === void 0 ? 0 : _props$value,
      _props$range = props.range,
      range = _props$range === void 0 ? [] : _props$range,
      _props$rangeKey = props.rangeKey,
      rangeKey = _props$rangeKey === void 0 ? 'value' : _props$rangeKey,
      title = props.title,
      disabled = props.disabled,
      children = props.children,
      onChange = props.onChange;

  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      open = _useState2[0],
      setOpen = _useState2[1];

  var _useState3 = useState(heightUnit * 2),
      _useState4 = _slicedToArray(_useState3, 2),
      initial = _useState4[0],
      setInitial = _useState4[1];

  var _useState5 = useState(0),
      _useState6 = _slicedToArray(_useState5, 2),
      start = _useState6[0],
      setStart = _useState6[1];

  var _useState7 = useState(initial),
      _useState8 = _slicedToArray(_useState7, 2),
      y = _useState8[0],
      setY = _useState8[1];

  var _useState9 = useState(0),
      _useState10 = _slicedToArray(_useState9, 2),
      columnHeight = _useState10[0],
      setColumnHeight = _useState10[1];

  var _useState11 = useState(false),
      _useState12 = _slicedToArray(_useState11, 2),
      isTransition = _useState12[0],
      setIsTransition = _useState12[1];

  useEffect(function () {
    setColumnHeight(range.length * heightUnit);
  }, [range]);
  useEffect(function () {
    if (!value || value <= 0) {
      setY(heightUnit * 2);
      return;
    }

    setY(-heightUnit * value + heightUnit * 2);
  }, [value]);

  var handleTap = function handleTap() {
    if (disabled) {
      return;
    }

    setOpen(function (state) {
      return !state;
    });
  };

  var handleCancel = function handleCancel() {
    setOpen(false);
  };

  var handleOK = function handleOK() {
    var v = Math.floor((y - heightUnit * 2) / -heightUnit);
    var e = {
      detail: {
        value: v
      }
    };
    onChange === null || onChange === void 0 ? void 0 : onChange(e);
    setOpen(false);
  };

  var handleTouchStart = function handleTouchStart(e) {
    var _e$preventDefault;

    (_e$preventDefault = e.preventDefault) === null || _e$preventDefault === void 0 ? void 0 : _e$preventDefault.call(e);
    setIsTransition(false);
    setStart(e.targetTouches[0].clientY);
  };

  var handleTouchMove = function handleTouchMove(e) {
    var _e$preventDefault2;

    (_e$preventDefault2 = e.preventDefault) === null || _e$preventDefault2 === void 0 ? void 0 : _e$preventDefault2.call(e);
    var diff = e.targetTouches[0].clientY - start;
    var updateY = diff + initial;

    if (updateY < -columnHeight + heightUnit * 2) {
      setY(-columnHeight + heightUnit * 2);
      return;
    }

    if (updateY > heightUnit * 3) {
      setY(heightUnit * 3);
      return;
    }

    setY(updateY);
  };

  var handleTouchEnd = function handleTouchEnd(e) {
    var _e$preventDefault3;

    (_e$preventDefault3 = e.preventDefault) === null || _e$preventDefault3 === void 0 ? void 0 : _e$preventDefault3.call(e);

    if (y < -columnHeight + heightUnit * 3) {
      setY(-columnHeight + heightUnit * 3);
      setInitial(-columnHeight + heightUnit * 3);
      setIsTransition(true);
      return;
    }

    if (y > heightUnit * 2) {
      setY(heightUnit * 2);
      setInitial(heightUnit * 2);
      setIsTransition(true);
      return;
    }

    if (y % heightUnit !== 0) {
      var r = Math.round(y / heightUnit) - Math.floor(y / heightUnit);

      if (r === 1) {
        var updateY = Math.ceil(y / heightUnit) * heightUnit;
        setY(updateY);
        setInitial(updateY);
        setIsTransition(true);
      } else {
        var _updateY = Math.floor(y / heightUnit) * heightUnit;

        setY(_updateY);
        setInitial(_updateY);
        setIsTransition(true);
      }

      return;
    }

    setInitial(y);
  };

  return /*#__PURE__*/React.createElement("div", {
    className: prefixCls
  }, /*#__PURE__*/React.createElement("div", {
    onClick: handleTap
  }, children), /*#__PURE__*/React.createElement(Popup, {
    position: "bottom",
    open: open,
    onClose: function onClose() {
      return setOpen(false);
    }
  }, /*#__PURE__*/React.createElement("div", {
    className: "".concat(prefixCls, "-container")
  }, /*#__PURE__*/React.createElement("div", {
    className: "".concat(prefixCls, "-toolbar")
  }, /*#__PURE__*/React.createElement("div", {
    className: "".concat(prefixCls, "-toolbar-cancel"),
    onClick: handleCancel
  }, "\u53D6\u6D88"), /*#__PURE__*/React.createElement("div", {
    className: "".concat(prefixCls, "-toolbar-tittle")
  }, title), /*#__PURE__*/React.createElement("div", {
    className: "".concat(prefixCls, "-toolbar-confirm"),
    onClick: handleOK
  }, "\u786E\u5B9A")), /*#__PURE__*/React.createElement("div", {
    className: "".concat(prefixCls, "-columns")
  }, /*#__PURE__*/React.createElement("div", {
    className: "".concat(prefixCls, "-column")
  }, /*#__PURE__*/React.createElement("ul", {
    className: "".concat(prefixCls, "-column-items"),
    onTouchStart: handleTouchStart,
    onTouchMove: handleTouchMove,
    onTouchEnd: handleTouchEnd // onMouseDown={handleTouchStart}
    // onMouseMove={handleTouchMove}
    // onMouseUp={handleTouchEnd}
    ,
    style: {
      transform: "translate3d(0, ".concat(y, "PX, 0)"),
      transition: "".concat(isTransition ? 'transform' : 'none', " 0.2s cubic-bezier(0.645, 0.045, 0.355, 1)")
    }
  }, range === null || range === void 0 ? void 0 : range.map(function (option, index) {
    return /*#__PURE__*/React.createElement("li", {
      key: index,
      className: "".concat(prefixCls, "-column-item")
    }, (option === null || option === void 0 ? void 0 : option[rangeKey]) || option);
  }))), /*#__PURE__*/React.createElement("div", {
    className: "".concat(prefixCls, "-selected")
  }), /*#__PURE__*/React.createElement("div", {
    className: "".concat(prefixCls, "-mask")
  })))));
};

export default Picker;
