// annar/web/images/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var React = _interopRequireWildcard(require("react"));

var _web = require("../web");

var _common = require("../common");

var prefixCls = (0, _common.getPrefixCls)('images');

var Images = function Images(props) {
  var _props$data = props.data,
      data = _props$data === void 0 ? [] : _props$data;

  var handleClickImage = function handleClickImage(index) {
    var urls = data;
    var current = index;

    if (typeof data[index] !== 'string') {
      urls = data.map(function (i) {
        return i.url;
      });
    }

    (0, _web.previewImage)({
      urls: urls,
      current: current,
      // @ts-ignore
      enablesavephoto: true,
      enableShowPhotoDownload: true
    });
  };

  return /*#__PURE__*/React.createElement(_web.View, {
    className: prefixCls
  }, data && data.map(function (item, index) {
    return /*#__PURE__*/React.createElement(_web.View, {
      key: item.key || index,
      className: "".concat(prefixCls, "-item"),
      onTap: function onTap() {
        return handleClickImage(index);
      }
    }, /*#__PURE__*/React.createElement(_web.Image, {
      mode: "widthFix",
      src: item.url || item
    }));
  }));
};

var _default = Images;
exports["default"] = _default;
