// annar/esm/action-sheet/index.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import React from 'react';
import { View } from 'remax/one';
import classNames from 'classnames';
import Popup from '../popup';
import Icon from '../icon';
import { getPrefixCls } from '../common';
var prefixCls = getPrefixCls('action-sheet');

var ActionSheet = function ActionSheet(props) {
  var open = props.open,
      title = props.title,
      actions = props.actions,
      _props$cancelText = props.cancelText,
      cancelText = _props$cancelText === void 0 ? 'Cancel' : _props$cancelText,
      cover = props.cover,
      _props$maskClosable = props.maskClosable,
      maskClosable = _props$maskClosable === void 0 ? true : _props$maskClosable,
      onCancel = props.onCancel,
      onChange = props.onChange;

  var handleCancel = function handleCancel(e) {
    onCancel === null || onCancel === void 0 ? void 0 : onCancel(e);
  };

  var handleChange = function handleChange(action, grid, e) {
    onChange === null || onChange === void 0 ? void 0 : onChange(action, grid, e);
  };

  var handleCloseMask = function handleCloseMask() {
    maskClosable && (onCancel === null || onCancel === void 0 ? void 0 : onCancel());
  };

  var renderAction = function renderAction(action) {
    return /*#__PURE__*/React.createElement(View, {
      key: action.value,
      className: classNames("".concat(prefixCls, "-action"), _defineProperty({}, "".concat(prefixCls, "-action-destructive"), action.type === 'destructive')),
      onTap: function onTap(e) {
        return handleChange(action, undefined, e);
      }
    }, action.text);
  };

  var renderGrid = function renderGrid(action) {
    var _action$text;

    return /*#__PURE__*/React.createElement(View, {
      key: action.value,
      className: classNames("".concat(prefixCls, "-grid"), _defineProperty({}, "".concat(prefixCls, "-action-destructive"), action.type === 'destructive'))
    }, /*#__PURE__*/React.createElement(View, {
      className: "".concat(prefixCls, "-grid-wrapper")
    }, (_action$text = action.text) === null || _action$text === void 0 ? void 0 : _action$text.map(function (item) {
      return /*#__PURE__*/React.createElement(View, {
        key: item.value,
        className: "".concat(prefixCls, "-grid-item"),
        onTap: function onTap(e) {
          return handleChange(item, action, e);
        }
      }, /*#__PURE__*/React.createElement(View, {
        className: "".concat(prefixCls, "-grid-item-icon")
      }, item.icon ? /*#__PURE__*/React.createElement(Icon, {
        type: item.icon,
        size: "48px",
        color: "rgba(0, 0, 0, 0.8)"
      }) : item.name), /*#__PURE__*/React.createElement(View, {
        className: "".concat(prefixCls, "-grid-item-name")
      }, item.name));
    })));
  };

  return /*#__PURE__*/React.createElement(View, {
    className: prefixCls
  }, /*#__PURE__*/React.createElement(Popup, {
    transparent: true,
    position: "bottom",
    open: open,
    onClose: handleCloseMask
  }, /*#__PURE__*/React.createElement(View, {
    className: classNames("".concat(prefixCls, "-container"), _defineProperty({}, "".concat(prefixCls, "-cover"), cover))
  }, /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-menu")
  }, title ? /*#__PURE__*/React.createElement(View, {
    className: classNames("".concat(prefixCls, "-action"), _defineProperty({}, "".concat(prefixCls, "-title"), true))
  }, title) : null, actions === null || actions === void 0 ? void 0 : actions.map(function (action) {
    return action.mode === 'grid' ? renderGrid(action) : renderAction(action);
  })), /*#__PURE__*/React.createElement(View, {
    className: "".concat(prefixCls, "-cancel"),
    onTap: handleCancel
  }, cancelText))));
};

export default ActionSheet;
