// annar/web/skeleton/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _web = require("../web");

var _classnames = _interopRequireDefault(require("classnames"));

var _common = require("../common");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var prefixCls = (0, _common.getPrefixCls)('skeleton');

var Skeleton = function Skeleton(props) {
  var _classNames7;

  var _props$title = props.title,
      title = _props$title === void 0 ? true : _props$title,
      titleColor = props.titleColor,
      backgroundColor = props.backgroundColor,
      active = props.active,
      _props$avatar = props.avatar,
      avatar = _props$avatar === void 0 ? false : _props$avatar,
      _props$loading = props.loading,
      loading = _props$loading === void 0 ? true : _props$loading,
      _props$image = props.image,
      image = _props$image === void 0 ? false : _props$image,
      _props$paragraph = props.paragraph,
      paragraph = _props$paragraph === void 0 ? {
    rows: 3,
    width: [80, 'auto', 200]
  } : _props$paragraph,
      _props$repetitions = props.repetitions,
      repetitions = _props$repetitions === void 0 ? 1 : _props$repetitions,
      space = props.space,
      customize = props.customize,
      fade = props.fade,
      style = props.style,
      children = props.children;

  var _React$useState = _react["default"].useState(false),
      _React$useState2 = (0, _slicedToArray2["default"])(_React$useState, 2),
      loadingEnd = _React$useState2[0],
      setLoadingEnd = _React$useState2[1];

  var _React$useState3 = _react["default"].useState(false),
      _React$useState4 = (0, _slicedToArray2["default"])(_React$useState3, 2),
      renderStart = _React$useState4[0],
      setRenderStart = _React$useState4[1];

  (0, _react.useEffect)(function () {
    if (fade) {
      if (!loading) {
        setTimeout(function () {
          setLoadingEnd(true);
        }, 300);
      }

      if (loading) {
        setLoadingEnd(false);
        setRenderStart(false);
      }
    }
  }, [loading]);
  (0, _react.useEffect)(function () {
    if (fade) {
      if (loadingEnd) {
        setTimeout(function () {
          setRenderStart(true);
        }, 100);
      }
    }
  }, [loadingEnd]);
  var rows = (0, _react.useMemo)(function () {
    return paragraph.rows ? (0, _toConsumableArray2["default"])(new Array(paragraph.rows).keys()) : [];
  }, [paragraph]);
  var rowsWidth = (0, _react.useMemo)(function () {
    if (typeof paragraph.width === 'number') {
      return (0, _toConsumableArray2["default"])(new Array(rows.length).fill("".concat(paragraph.width, "px")));
    }

    if (typeof paragraph.width === 'string') {
      return (0, _toConsumableArray2["default"])(new Array(rows.length).fill(paragraph.width));
    }

    if (Array.isArray(paragraph.width)) {
      var _paragraph$width;

      return (_paragraph$width = paragraph.width) === null || _paragraph$width === void 0 ? void 0 : _paragraph$width.map(function (i) {
        if (typeof i === 'number') {
          return "".concat(i, "px");
        } else if (typeof i === 'string') {
          return i;
        } else {
          return 'auto';
        }
      });
    }
  }, [paragraph]);

  var renderImage = function renderImage(visible) {
    var _classNames;

    return visible ? /*#__PURE__*/_react["default"].createElement(_web.View, {
      className: (0, _classnames["default"])((_classNames = {}, (0, _defineProperty2["default"])(_classNames, "".concat(prefixCls, "-image"), true), (0, _defineProperty2["default"])(_classNames, "".concat(prefixCls, "-active"), active), _classNames))
    }) : null;
  };

  var renderAvatar = function renderAvatar(visible) {
    var _classNames2;

    return visible ? /*#__PURE__*/_react["default"].createElement(_web.View, {
      className: (0, _classnames["default"])((_classNames2 = {}, (0, _defineProperty2["default"])(_classNames2, "".concat(prefixCls, "-avatar"), true), (0, _defineProperty2["default"])(_classNames2, "".concat(prefixCls, "-active"), active), _classNames2))
    }) : null;
  };

  var renderTitle = function renderTitle(visible) {
    var _classNames3;

    return visible ? /*#__PURE__*/_react["default"].createElement(_web.View, {
      className: (0, _classnames["default"])((_classNames3 = {}, (0, _defineProperty2["default"])(_classNames3, "".concat(prefixCls, "-title"), true), (0, _defineProperty2["default"])(_classNames3, "".concat(prefixCls, "-active"), active), _classNames3)),
      style: {
        backgroundColor: titleColor
      }
    }) : null;
  };

  var renderParagraph = function renderParagraph() {
    return rows.map(function (item, index) {
      var _classNames4;

      return /*#__PURE__*/_react["default"].createElement(_web.View, {
        key: item,
        className: (0, _classnames["default"])((_classNames4 = {}, (0, _defineProperty2["default"])(_classNames4, "".concat(prefixCls, "-row"), true), (0, _defineProperty2["default"])(_classNames4, "".concat(prefixCls, "-active"), active), _classNames4)),
        style: {
          width: rowsWidth === null || rowsWidth === void 0 ? void 0 : rowsWidth[index]
        }
      });
    });
  };

  var renderSkeleton = function renderSkeleton() {
    var skeleton = (0, _toConsumableArray2["default"])(new Array(repetitions).keys());
    return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, skeleton === null || skeleton === void 0 ? void 0 : skeleton.map(function (i, index) {
      return /*#__PURE__*/_react["default"].createElement(_web.View, {
        key: i,
        className: "".concat(prefixCls, "-item"),
        style: {
          marginBottom: index !== skeleton.length - 1 ? space : 0
        }
      }, customize ? customize : /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, renderImage(image), renderAvatar(avatar), renderTitle(title), renderParagraph()));
    }));
  };

  if (fade) {
    var _classNames6;

    return /*#__PURE__*/_react["default"].createElement(_web.View, {
      className: prefixCls
    }, !loadingEnd ? /*#__PURE__*/_react["default"].createElement(_web.View, {
      className: (0, _classnames["default"])("".concat(prefixCls, "-placeholder"), (0, _defineProperty2["default"])({}, "".concat(prefixCls, "-hidden"), !loading)),
      style: _objectSpread({
        backgroundColor: backgroundColor
      }, style)
    }, renderSkeleton()) : /*#__PURE__*/_react["default"].createElement(_web.View, {
      className: (0, _classnames["default"])((_classNames6 = {}, (0, _defineProperty2["default"])(_classNames6, "".concat(prefixCls, "-container"), true), (0, _defineProperty2["default"])(_classNames6, "".concat(prefixCls, "-container-active"), renderStart), _classNames6))
    }, children));
  }

  return /*#__PURE__*/_react["default"].createElement(_web.View, {
    className: prefixCls
  }, loading ? /*#__PURE__*/_react["default"].createElement(_web.View, {
    className: "".concat(prefixCls, "-placeholder"),
    style: _objectSpread({
      backgroundColor: backgroundColor
    }, style)
  }, renderSkeleton()) : /*#__PURE__*/_react["default"].createElement(_web.View, {
    className: (0, _classnames["default"])((_classNames7 = {}, (0, _defineProperty2["default"])(_classNames7, "".concat(prefixCls, "-container"), true), (0, _defineProperty2["default"])(_classNames7, "".concat(prefixCls, "-container-active"), true), _classNames7))
  }, children));
};

var _default = Skeleton;
exports["default"] = _default;
