// annar/esm/filter/item.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import React, { useImperativeHandle, useState } from 'react';
import { View, Text } from 'remax/one';
import classNames from 'classnames';
import { ScrollView } from '../one';
import Dropdown from '../dropdown';
import Popup from '../popup';
import { getPrefixCls } from '../common';
var prefixCls = getPrefixCls('filter-item');

var Item = function Item(props, ref) {
  var _classNames, _options$find;

  var title = props.title,
      onTap = props.onTap,
      children = props.children,
      onShowMask = props.onShowMask,
      value = props.value,
      _props$options = props.options,
      options = _props$options === void 0 ? [] : _props$options,
      onChange = props.onChange,
      activeColor = props.activeColor,
      disabled = props.disabled;

  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      open = _useState2[0],
      setOpen = _useState2[1];

  useImperativeHandle(ref, function () {
    return {
      toggle: function toggle() {
        setOpen(function (state) {
          if (state) {
            setTimeout(function () {
              onShowMask === null || onShowMask === void 0 ? void 0 : onShowMask(false);
            });
          } else {
            setTimeout(function () {
              onShowMask === null || onShowMask === void 0 ? void 0 : onShowMask(true);
            });
          }

          return !state;
        });
      },
      open: function open() {
        setOpen(true);
      },
      close: function close() {
        setOpen(false);
      }
    };
  }, []);

  var handleTap = function handleTap() {
    if (disabled) {
      return;
    }

    setOpen(function (state) {
      return !state;
    });
    onTap === null || onTap === void 0 ? void 0 : onTap(!open);
  };

  var handleChange = function handleChange(e) {
    onChange === null || onChange === void 0 ? void 0 : onChange(e);
    setOpen(false);
    onShowMask === null || onShowMask === void 0 ? void 0 : onShowMask(false);
  };

  return /*#__PURE__*/React.createElement(View, {
    className: prefixCls
  }, /*#__PURE__*/React.createElement(View, {
    style: {
      backgroundColor: '#FDFFFD'
    }
  }, /*#__PURE__*/React.createElement(View, {
    className: classNames((_classNames = {}, _defineProperty(_classNames, "".concat(prefixCls, "-label"), true), _defineProperty(_classNames, "".concat(prefixCls, "-label-disabled"), disabled), _classNames)),
    onTap: handleTap
  }, /*#__PURE__*/React.createElement(Text, {
    className: "".concat(prefixCls, "-label-text")
  }, (options === null || options === void 0 ? void 0 : (_options$find = options.find(function (option) {
    return option.value === value;
  })) === null || _options$find === void 0 ? void 0 : _options$find.text) || title), open ? /*#__PURE__*/React.createElement(Text, {
    className: "".concat(prefixCls, "-label-chevronUp")
  }) : /*#__PURE__*/React.createElement(Text, {
    className: "".concat(prefixCls, "-label-chevronDown")
  }))), /*#__PURE__*/React.createElement(Popup, {
    position: "bottom",
    square: true,
    curve: "ease",
    mask: false,
    open: open,
    style: {
      position: 'absolute',
      zIndex: -1,
      transform: open ? 'translate3d(0, 100%, 0)' : 'translate3d(0, 0, 0)'
    }
  }, /*#__PURE__*/React.createElement(ScrollView, {
    scrollY: true,
    className: classNames(_defineProperty({}, "".concat(prefixCls, "-value"), true))
  }, children ? children : /*#__PURE__*/React.createElement(Dropdown, {
    value: value,
    options: options,
    onChange: handleChange,
    activeColor: activeColor
  }))));
};

export default /*#__PURE__*/React.forwardRef(Item);
