// annar/web/steps/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var React = _interopRequireWildcard(require("react"));

var _web = require("../web");

var _classnames = _interopRequireDefault(require("classnames"));

var _icon = _interopRequireDefault(require("../icon"));

var _common = require("../common");

var prefixCls = (0, _common.getPrefixCls)('steps');

var Steps = function Steps(props) {
  var _props$direction = props.direction,
      direction = _props$direction === void 0 ? 'horizontal' : _props$direction,
      _props$current = props.current,
      current = _props$current === void 0 ? 0 : _props$current,
      steps = props.steps,
      status = props.status;
  var statusIconType = status === 'error' ? 'roundclosefill' : 'roundcheckfill';
  var statusIconColor = status === 'error' ? '#f5222d' : '#1890FF';

  var getTitle = function getTitle(item, index) {
    if (index === current) {
      if (status === 'error') {
        return item.errorTitle || item.title;
      }
    }

    return item.title;
  };

  if (direction === 'vertical') {
    return /*#__PURE__*/React.createElement(_web.View, {
      className: prefixCls
    }, steps === null || steps === void 0 ? void 0 : steps.map(function (item, index) {
      return /*#__PURE__*/React.createElement(_web.View, {
        key: index,
        className: "".concat(prefixCls, "-step")
      }, index === 0 ? /*#__PURE__*/React.createElement(_web.View, {
        className: "".concat(prefixCls, "-circle_current")
      }, /*#__PURE__*/React.createElement(_web.Text, {
        className: "".concat(prefixCls, "-circle_current-icon")
      }, /*#__PURE__*/React.createElement(_icon["default"], {
        type: "roundcheckfill",
        size: "40px",
        color: "#1890FF"
      }))) : /*#__PURE__*/React.createElement(_web.View, {
        className: "".concat(prefixCls, "-circle")
      }), /*#__PURE__*/React.createElement(_web.View, {
        className: "".concat(prefixCls, "-line")
      }), /*#__PURE__*/React.createElement(_web.View, {
        className: "".concat(prefixCls, "-step-header")
      }, /*#__PURE__*/React.createElement(_web.View, {
        className: "".concat(prefixCls, "-step-header-title")
      }, item.title), /*#__PURE__*/React.createElement(_web.View, {
        className: "".concat(prefixCls, "-step-header-extra")
      }, item.extra)), /*#__PURE__*/React.createElement(_web.View, {
        className: "".concat(prefixCls, "-step-description")
      }, item.description));
    }));
  }

  return /*#__PURE__*/React.createElement(_web.View, {
    className: "".concat(prefixCls, "_horizontal")
  }, steps === null || steps === void 0 ? void 0 : steps.map(function (item, index) {
    var _classNames;

    return /*#__PURE__*/React.createElement(_web.View, {
      key: index,
      className: (0, _classnames["default"])((_classNames = {}, (0, _defineProperty2["default"])(_classNames, "".concat(prefixCls, "_horizontal-step"), true), (0, _defineProperty2["default"])(_classNames, "".concat(prefixCls, "_horizontal-step-finish"), index <= current - 1), _classNames))
    }, /*#__PURE__*/React.createElement(_web.View, {
      className: "".concat(prefixCls, "_horizontal-step-container")
    }, /*#__PURE__*/React.createElement(_web.View, {
      className: "".concat(prefixCls, "_horizontal-step-container-icon")
    }, index < current ? /*#__PURE__*/React.createElement(_icon["default"], {
      type: "roundcheckfill",
      size: "40px",
      color: "#1890FF"
    }) : null, index === current ? /*#__PURE__*/React.createElement(_icon["default"], {
      type: statusIconType,
      size: "40px",
      color: statusIconColor
    }) : null, index > current ? /*#__PURE__*/React.createElement(_web.View, {
      className: "".concat(prefixCls, "_horizontal-step-container-icon-default")
    }, /*#__PURE__*/React.createElement(_web.View, null)) : null), /*#__PURE__*/React.createElement(_web.View, {
      className: "".concat(prefixCls, "_horizontal-step-container-content")
    }, /*#__PURE__*/React.createElement(_web.View, {
      className: "".concat(prefixCls, "_horizontal-step-container-content-title")
    }, getTitle(item, index)))));
  }));
};

var _default = Steps;
exports["default"] = _default;
